http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/RuntimeServiceInterp.scala

```diff
@@ -307,11 +307,10 @@ class RuntimeServiceInterp[F[_]: Parallel](config: RuntimeServiceConfig,
                   F.unit)
               )
               _ <- RuntimeConfigQueries.updatePersistentDiskId(runtime.runtimeConfigId, None, ctx.now).transaction
-            } yield {
-              if (req.deleteDisk)
-                Some(diskId)
-              else None
-            }
+              res <- if (req.deleteDisk)
+                persistentDiskQuery.updateStatus(diskId, DiskStatus.Deleting, ctx.now).transaction.as(Some(diskId))
```

Comment:
this is the real fix...previously we weren't updating disk status properly, hence UI is reusing the same disk. But UI won't try to re-use the same disk if disk is marked as `Deleting`

Reply:
Makes sense -- so the disk deletion will be done by Back Leo, and back Leo will then transition the disk to `Deleted`?

Reply:
yep https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala#L222

Reply:
Tested in fiab....with dev code, if I create a runtime immediately after deletion, UI try to use the previously created disk...With this PR (disk status is updated to `Deleting` once delete runtime call finishes), when I create a runtime immediately after deletion, UI will create a new disk name. cc @panentheos  in case I'm misunderstanding UI behavior